flyteplugins/go/tasks/plugins/webapi/agent/plugin.go

```diff
@@ -299,21 +316,82 @@ func getFinalContext(ctx context.Context, operation string, agent *Agent) (conte
 	if timeout == 0 {
 		return ctx, func() {}
 	}
+
 	return context.WithTimeout(ctx, timeout)
 }
 
+func initializeAgentRegistry(cfg *Config, connectionCache map[*Agent]*grpc.ClientConn, getAgentMetadataClientFunc GetAgentMetadataClientFunc) (map[string]*Agent, error) {
+	agentRegistry := make(map[string]*Agent)
+	var agentDeployments []*Agent
+
+	// Ensure that the old configuration is backward compatible
+	for taskType, agentID := range cfg.AgentForTaskTypes {
+		agentRegistry[taskType] = cfg.Agents[agentID]
+	}
+
+	if len(cfg.DefaultAgent.Endpoint) != 0 {
+		agentDeployments = append(agentDeployments, &cfg.DefaultAgent)
+	}
+	agentDeployments = append(agentDeployments, maps.Values(cfg.Agents)...)
+	for _, agentDeployment := range agentDeployments {
+		client, err := getAgentMetadataClientFunc(context.Background(), agentDeployment, connectionCache)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to agent [%v] with error: [%v]", agentDeployment, err)
```

Comment:
Will this fail flytepropeller startup? If so, I think we need to be more resilient and not fail the whole propeller due to a single misbehaving agent deployment.

Reply:
We could issue a warning and move on to the next deployment. When routing task later we will also need to be more resilient and fail if a mapping is not found.